tests/cli/test_conda_argparse.py

```diff
@@ -133,9 +134,15 @@ def test_sorted_commands_in_error(capsys):
     sp.add_parser("b")
     try:
         p.parse_args(["d"])
+    # Python < 3.12
     except SystemExit:
         stderr = capsys.readouterr().err
         # ...but the suggestions here are sorted
         assert "invalid choice: 'd' (choose from 'a', 'b', 'c')" in stderr
+    # Python >= 3.12
+    except ArgumentError:
+        stderr = capsys.readouterr().err
+        # ...but the suggestions here are sorted
+        assert "invalid choice: 'd' (choose from a, b, c)" in stderr
```

Comment:
Looks like this is not being raised in Python 3.12 CI either. Maybe it's better to catch both and assert that one of the options is there?

Something like:

```python
assert "invalid choice: 'd' (choose from a, b, c)" in stderr or "invalid choice: 'd' (choose from 'a', 'b', 'c')" in stderr
```

Reply:
Ah, I see the issue, the exception never changed, it was an `ArgumentError` and still is (`parse_args` intercepts all exceptions and raises a `SystemExit`) ([3.9](https://github.com/python/cpython/blob/3.9/Lib/argparse.py#L2501-L2507), [3.10](https://github.com/python/cpython/blob/3.10/Lib/argparse.py#L2513-L2519), [3.11](https://github.com/python/cpython/blob/3.11/Lib/argparse.py#L2557-L2563), [3.12](https://github.com/python/cpython/blob/3.12/Lib/argparse.py#L2563-L2573))

Only the choices rendering changed (`str` instead of `repr`): https://github.com/python/cpython/pull/117766

Reply:
Yep, that's what I wanted to do as well, and then the incident with libmamba happened, thanks @kenodegard for updating the PR!